gcd/src/main/java/org/spine3/server/storage/datastore/DatastoreStorageFactory.java

```diff
@@ -213,15 +226,19 @@ private Builder() {
          * @return self for method chaining
          */
         public Builder setDatastore(Datastore datastore) {
-            this.datastore = checkNotNull(datastore);
+            checkHasNoNamespace(datastore);
```

Comment:
This should only apply in case the factory is multitenant. So I would move that to `build()`. And bring back `checkNotNull(datastore)`.